src/clientHandler.ts

```diff
@@ -12,298 +10,177 @@ import {
-    }
-    if (terraformLogFilePath.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { terraformLogFilePath });
-    }
-    if (rootModulePaths.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { rootModulePaths });
-    }
-    if (excludeModulePaths.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { excludeModulePaths });
-    }
-    if (ignoreDirectoryNames.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { ignoreDirectoryNames });
-    }
+    const documentSelector: DocumentSelector = [
+      { scheme: 'file', language: 'terraform' },
+      { scheme: 'file', language: 'terraform-vars' },
+    ];
 
-    const executable: Executable = {
-      command: cmd,
-      args: serverArgs,
-      options: {},
-    };
-    const serverOptions: ServerOptions = {
-      run: executable,
-      debug: executable,
-    };
     const clientOptions: LanguageClientOptions = {
       documentSelector: documentSelector,
-      workspaceFolder: wsFolder,
       initializationOptions: initializationOptions,
       initializationFailedHandler: (error) => {
         this.reporter.sendTelemetryException(error);
         return false;
       },
-      outputChannel: outputChannel,
+      outputChannel: this.outputChannel,
       revealOutputChannelOn: RevealOutputChannelOn.Never,
     };
 
-    const client = new LanguageClient(id, name, serverOptions, clientOptions);
+    const id = `terraform`;
+    const client = new LanguageClient(id, serverOptions, clientOptions);
 
+    const enableReferenceCountCodeLens: boolean = config('terraform').get('enableReferenceCountCodeLens');
     if (enableReferenceCountCodeLens) {
       client.registerFeature(new ShowReferencesFeature(client));
     }
 
     client.onDidChangeState((event) => {
+      console.log(`Client: ${State[event.oldState]} --> ${State[event.newState]}`);
       if (event.newState === State.Stopped) {
-        clients.delete(location);
         this.reporter.sendTelemetryEvent('stopClient');
       }
     });
 
     return { commandPrefix, client };
   }
 
-  public async stopClients(folders?: string[]): Promise<void[]> {
-    const promises: Promise<void>[] = [];
+  private getServerOptions() {
+    const cmd = this.lsPath.resolvedPathToBinary();
+    const serverArgs: string[] = config('terraform').get('languageServer.args');
+    const executable: Executable = {
+      command: cmd,
+      args: serverArgs,
+      options: {},
+    };
+    const serverOptions: ServerOptions = {
+      run: executable,
+      debug: executable,
+    };
+    return serverOptions;
+  }
 
-    if (this.supportsMultiFolders) {
-      promises.push(this.stopClient(MULTI_FOLDER_CLIENT));
-      return Promise.all(promises);
-    }
+  private getInitializationOptions(commandPrefix: string) {
+    const rootModulePaths: string[] = config('terraform-ls').get('rootModules');
```

Comment:
Here too and maybe add a sensible default for the following `length` checks
```suggestion
    const rootModulePaths = config('terraform-ls').get<string[]>('rootModules', []);
```